### PR TITLE
Containers Summary test + Updated StatusBox object

### PR DESCRIPTION
**Pull request by pavelzag**
# Purpose or Intent

This tests checks status boxes in the Containers Overview menu and verifies it shows the correct data for:
Pods, Containers and Nodes

{{pytest: cfme/tests/containers/test_containers_summary.py -v --use-provider container }}

### CODE REVIEW

**Comment by pavelzag** [comment]
Replaces https://github.com/ManageIQ/integration_tests/pull/3341 and https://github.com/ManageIQ/integration_tests/pull/3362

**Comment by dajoRH** [comment]
Lint report for commit ea928f70dad5837bf1c944272ebfd5a6347c8980:

Everything seems all right :smile: .
*CFME QE Bot*

**Comment by pavelzag** [comment]
This run fails probably due to a known bug
https://bugzilla.redhat.com/show_bug.cgi?id=1369359

File: cfme/tests/containers/test_containers_summary.py [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+        in the Containers widget in the
+       Overview menu
+       Steps:
+           * Goes to Compute --> Containers --> Overview
+           * Checks how many Containers are shown in the Containers Widget
+           * Goes to Containers summary page and checks how many Containers are shown there.
+           * Checks the amount is equal
+       """
+    sel.force_navigate('container_dashboard')
+    time.sleep(2)
```
Comment (nellyc) [inline cfme/tests/containers/test_containers_summary.py]: why is this sleep needed? cant we wait for some object?

Comment (pavelzag) [inline cfme/tests/containers/test_containers_summary.py]: This is the solution at the moment. We will add a better solution soon

**Comment by dajoRH** [comment]
Lint report for commit 90877af9a73a6d0002d277c7f6c645ea3e0d374e:

Everything seems all right from lint perspective :smile: .

Commit inspection report:

**No commit flaws detected**.
<mergeable/>

*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit f342b28c3dd2296c6bf8517d5446453a9b008264:

Everything seems all right from lint perspective :smile: .

Commit inspection report:

**No commit flaws detected**.
<mergeable/>

*CFME QE Bot*

File: cfme/web_ui/__init__.py [hunk 2]
```diff
@@ -144,8 +144,8 @@ class Region(Pretty):
     Note:
 
         When specifying a region title, omit the "Cloudforms Management Engine: " or "ManageIQ: "
-        prefix. They are included on every page, and different for the two versions of the
-        appliance, and :py:meth:`is_displayed` strips them off before checking for equality.
+        prefix. They're included on every page, and different for the two versions of the appliance,
+        and :py:meth:`is_displayed` strips them off before checking for equality.
```
Comment (jkrocil) [inline cfme/web_ui/__init__.py]: This is incorrect merge I believe.

Comment (pavelzag) [inline cfme/web_ui/__init__.py]: Hmm yea, that's a weird one

Comment (jkrocil) [inline cfme/web_ui/__init__.py]: Once you fix this, I can merge it.

Comment (pavelzag) [inline cfme/web_ui/__init__.py]: @mfalesni I'd like to remove this change that somehow was introduced in order to pass the PR.
Not sure how it got there and how to remove it

Comment (mfalesni) [inline cfme/web_ui/__init__.py]: If you add a commit that reverts the changes and then squash it, it will disappear.

Comment (jkrocil) [inline cfme/web_ui/__init__.py]: Or just undo the change and amend the previous commit.

**Comment by pavelzag** [comment]
@jkrocil fixed

**Comment by dajoRH** [comment]
Lint report for commit b069a072f25d456d04fca7891af9cf23b1af4f56:

Everything seems all right from lint perspective :smile: .

Commit inspection report:

**No commit flaws detected**.
<mergeable/>

*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit b9749bc0c65e466b2cb6b3a5b0910de0a5c98d47:

Everything seems all right from lint perspective :smile: .

Commit inspection report:

**No commit flaws detected**.
<mergeable/>

*CFME QE Bot*

**Comment by jdemon519** [comment]
Approve +1

File: cfme/tests/containers/test_containers_summary.py [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+    testgen.container_providers, scope='function')
+
+
+# container_dashboard
+# CMP-9827 # CMP-9826 # CMP-9825 # CMP-9824 # CMP-9823 # CMP-9822 # CMP-9821 # CMP-9820
+SUMMARY_OBJECTS = \
+    ['routes', 'projects', 'images', 'registries', 'services', 'containers', 'pods', 'nodes']
+
+
+def test_containers_summary_objects(provider):
```
Comment (nellyc) [inline cfme/tests/containers/test_containers_summary.py]: i think it will be better to split it into multiple tests

File: cfme/web_ui/__init__.py [hunk 4]
```diff
@@ -3639,9 +3639,11 @@ class StatusBox(object):
...
     def __init__(self, name):
-        self.name = name
+        self.name = str(name).lower()
 
     def value(self):
-        return sel.element(
-            '//span[contains(@class, "card-pf-aggregate-status-count")]'
-            '/../../span[contains(., "{}")]/span'.format(self.name)).text
+        elem_text = sel.text(
+            "//span[contains(@class,'card-pf-aggregate-status-count')]"
```
Comment (nellyc) [inline cfme/web_ui/__init__.py]: automation id

File: cfme/tests/containers/test_containers_summary.py (hunk 3 above)
Comment (pavelzag) [inline cfme/tests/containers/test_containers_summary.py]: It started split but I saw it's cleaner and runs much much faster when combined

File: cfme/tests/containers/test_containers_summary.py [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+        in the widgets in the
+       Overview menu
+       Steps:
+           * Goes to Compute --> Containers --> Overview
+           * Checks how many objects are shown in the selected widget
+           * Goes to Containers summary page and checks how many objects are shown there.
+           * Checks the amount is equal
+       """
+    sel.force_navigate('container_dashboard')
+    sel.force_navigate('containers_provider', context={'provider': provider})
```
Comment (jkrocil) [inline cfme/tests/containers/test_containers_summary.py]: Navigating inside a test like this is strongly discouraged.

Comment (jkrocil) [inline cfme/tests/containers/test_containers_summary.py]: More so when the framework will navigate there on its own. You don't need to call either of those. When you do `provider.summary.anything` it will automatically call `provider.load_details()`, see here: https://github.com/ManageIQ/integration_tests/blob/master/cfme/common/__init__.py#L156

File: cfme/tests/containers/test_containers_summary.py [hunk 6]
```diff
@@ -0,0 +1,65 @@
...
+    projects_val = provider.summary.relationships.projects.value
+    images_val = provider.summary.relationships.images.value
+    image_registries_val = provider.summary.relationships.image_registries.value
+    services_val = provider.summary.relationships.services.value
+    containers_val = provider.summary.relationships.containers.value
+    pods_val = provider.summary.relationships.pods.value
+    nodes_val = provider.summary.relationships.nodes.value
+    for summary_object in SUMMARY_OBJECTS:
+        sel.force_navigate('container_dashboard')
+        time.sleep(2)
```
Comment (jkrocil) [inline cfme/tests/containers/test_containers_summary.py]: `time.sleep()` is also discouraged. It may work for you but if the test is ran on a machine that is on a different network, 2 seconds might not be enough and you will see cryptic errors...

File: cfme/tests/containers/test_containers_summary.py [hunk 7]
```diff
@@ -0,0 +1,65 @@
...
+    pytest.mark.usefixtures('has_no_container_providers'),
+    pytest.mark.usefixtures('setup_provider'),
+    pytest.mark.tier(1)]
+pytest_generate_tests = testgen.generate(
+    testgen.container_providers, scope='function')
+
+
+# container_dashboard
+# CMP-9827 # CMP-9826 # CMP-9825 # CMP-9824 # CMP-9823 # CMP-9822 # CMP-9821 # CMP-9820
+SUMMARY_OBJECTS = \
```
Comment (jkrocil) [inline cfme/tests/containers/test_containers_summary.py]: These are not summary objects... these are `container_object_types` or something similar, but not objects.

File: cfme/tests/containers/test_containers_summary.py [hunk 8]
```diff
@@ -0,0 +1,65 @@
...
+        elif summary_object == 'registries':
+            cont_val = image_registries_val
+        elif summary_object == 'services':
+            cont_val = services_val
+        elif summary_object == 'containers':
+            cont_val = containers_val
+        elif summary_object == 'pods':
+            cont_val = pods_val
+        elif summary_object == 'nodes':
+            cont_val = nodes_val
```
Comment (jkrocil) [inline cfme/tests/containers/test_containers_summary.py]: Ideally, container core, e.g. `cfme/containers/__init__.py` or `cfme/containers/dashboard.py` should have a function to fetch all these values by type. So that you can get rid of all these ifs here and also the `force_navigate` call and what you would do instead is:
```
prov_ui_values = dict()
# Load all values from provider page
for obj_type in container_object_types:
    prov_ui_values[obj_type] = getattr(provider.summary, obj_type).value

from cfme.containers.dashboard import get_object_count
# Check that all values on the dashboard correspond with provider values
for obj_type in container_object_types:
    get_object_count(obj_type) == prov_ui_values[obj_type]
```
the code above is not tested, but it should work.

Also, once you have multiple providers in CFME, this will obviously __all__ fail.

Comment (jkrocil) [inline cfme/tests/containers/test_containers_summary.py]: Now I realize the 6 lines of code above is actually your whole test really.

File: cfme/tests/containers/test_containers_summary.py (hunk 5 above)
Comment (pavelzag) [inline cfme/tests/containers/test_containers_summary.py]: I removed those

File: cfme/tests/containers/test_containers_summary.py (hunk 7 above)
Comment (pavelzag) [inline cfme/tests/containers/test_containers_summary.py]: Renamed

File: cfme/tests/containers/test_containers_summary.py (hunk 6 above)
Comment (pavelzag) [inline cfme/tests/containers/test_containers_summary.py]: We are working on a better solution but for the time being we'd really like to push this test. It's a bad solution but it's better than nothing at the moment

File: cfme/tests/containers/test_containers_summary.py (hunk 8 above)
Comment (pavelzag) [inline cfme/tests/containers/test_containers_summary.py]: This doesn't work.. I am getting: 
prov_ui_values[obj_type] = getattr(provider.summary, obj_type).value
E           AttributeError: 'Summary' object has no attribute 'routes'

I am counting amount of objects in StatusBox with a function I added there

Comment (pavelzag) [inline cfme/tests/containers/test_containers_summary.py]: Also, I see no such thing as dashboard under cfme.containers..

Comment (jkrocil) [inline cfme/tests/containers/test_containers_summary.py]: @pavelzag You would have to create the `cfme/containers/dashboard.py` module.
As for `'Summary' object has no attribute 'routes'` - my bad. It has to be `getattr(provider.summary.relationships, obj_type).value` or possibly `.text_value` or something similar but the main point was you can make it shorter and cleaner.

Comment (jkrocil) [inline cfme/tests/containers/test_containers_summary.py]: `get_object_count` is the function that would be defined inside of `__init__.py` or rather `dashboard.py` and fetch the data from the dashboard. It has to be created, it does not exist anywhere at this point.

Comment (pavelzag) [inline cfme/tests/containers/test_containers_summary.py]: I was advised to add this functionality under StatusBox in web_ui/__init__.py at first. I don't remember who told me to do that. Either Milan or Pete. Are you sure it's the right place?

Comment (pavelzag) [inline cfme/tests/containers/test_containers_summary.py]: There's a dashboard.py under cfme. Are we talking about this one?

Comment (pavelzag) [inline cfme/tests/containers/test_containers_summary.py]: Also, this will not work because there's a difference between labels in the overview menu and the labels under the provider menu. For instance Registries in Overview menu are called Image Registries in provider. Images = Container Images, Services = Container Services. There's a mess in the UI and it leaks to the tests

**Comment by dajoRH** [comment]
Lint report for commit 05459ed7ff1c140521544a28cc4b3e48cb32a04c:

`cfme/tests/containers/test_containers_summary.py`:
- :bangbang: Line 21:101: **E501** *line too long (119 > 100 characters)*
- :red_circle: Line 35:5: **F841** *local variable 'routes_val' is assigned to but never used*
- :red_circle: Line 36:5: **F841** *local variable 'projects_val' is assigned to but never used*
- :red_circle: Line 37:5: **F841** *local variable 'container_images_val' is assigned to but never used*
- :red_circle: Line 38:5: **F841** *local variable 'image_registries_val' is assigned to but never used*
- :red_circle: Line 39:5: **F841** *local variable 'container_services_val' is assigned to but never used*
- :red_circle: Line 40:5: **F841** *local variable 'containers_val' is assigned to but never used*
- :red_circle: Line 41:5: **F841** *local variable 'pods_val' is assigned to but never used*
- :red_circle: Line 42:5: **F841** *local variable 'nodes_val' is assigned to but never used*

Please, rectify these issues :smirk: .

Commit inspection report:

**No commit flaws detected**.
You have multiple commits in the PR. When you want your PR to be merged, please squash them. Do not squash them when working on the pull request as this allows to see the progress!

*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit 6fe2da4937c6d9bef5c043144e561866326e1cc8:

Everything seems all right from lint perspective :smile: .

Commit inspection report:

**No commit flaws detected**.
You have multiple commits in the PR. When you want your PR to be merged, please squash them. Do not squash them when working on the pull request as this allows to see the progress!

*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit 002c2a824d8e0b795f79f81627a4b3dc1c20f401:

Everything seems all right from lint perspective :smile: .

Commit inspection report:

**No commit flaws detected**.
You have multiple commits in the PR. When you want your PR to be merged, please squash them. Do not squash them when working on the pull request as this allows to see the progress!

*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit eaed8573c66f5bd9b505c170b4c2529eaa44d1ad:

`cfme/tests/containers/test_containers_summary.py`:
- :bangbang: Line 21:101: **E501** *line too long (119 > 100 characters)*
- :red_circle: Line 35:5: **F841** *local variable 'routes_val' is assigned to but never used*
- :red_circle: Line 36:5: **F841** *local variable 'projects_val' is assigned to but never used*
- :red_circle: Line 37:5: **F841** *local variable 'container_images_val' is assigned to but never used*
- :red_circle: Line 38:5: **F841** *local variable 'image_registries_val' is assigned to but never used*
- :red_circle: Line 39:5: **F841** *local variable 'container_services_val' is assigned to but never used*
- :red_circle: Line 40:5: **F841** *local variable 'containers_val' is assigned to but never used*
- :red_circle: Line 41:5: **F841** *local variable 'pods_val' is assigned to but never used*
- :red_circle: Line 42:5: **F841** *local variable 'nodes_val' is assigned to but never used*

Please, rectify these issues :smirk: .

Commit inspection report:

**No commit flaws detected**.

*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit 26b051ff93298399cd39f76ef072235cb583a36c:

`cfme/tests/containers/test_containers_summary.py`:
- :heavy_exclamation_mark: Line 45:1: **W391** *blank line at end of file*

Please, rectify these issues :smirk: .

Commit inspection report:

**No commit flaws detected**.

*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit 5881e9c842999cf39142ce72d94e9eb1de07f2bb:

Everything seems all right from lint perspective :smile: .

Commit inspection report:

**No commit flaws detected**.
<mergeable/>

*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit a4a44080d410d8d4221c5299f2a6448039c06bb7:

Everything seems all right from lint perspective :smile: .

Commit inspection report:

**No commit flaws detected**.
<mergeable/>

*CFME QE Bot*

**Comment by pavelzag** [comment]
@jkrocil @mfalesni Hey guys, can you check again and merge if possible?

**Comment by pavelzag** [comment]
@jkrocil Bump
